domain/lib/src/usecases/download_file/download_file_interactor.dart

```diff
@@ -0,0 +1,54 @@
+// LinShare is an open source filesharing software, part of the LinPKI software
+// suite, developed by Linagora.
+//
+// Copyright (C) 2020 LINAGORA
+//
+// This program is free software: you can redistribute it and/or modify it under the
+// terms of the GNU Affero General Public License as published by the Free Software
+// Foundation, either version 3 of the License, or (at your option) any later version,
+// provided you comply with the Additional Terms applicable for LinShare software by
+// Linagora pursuant to Section 7 of the GNU Affero General Public License,
+// subsections (b), (c), and (e), pursuant to which you must notably (i) retain the
+// display in the interface of the “LinShare™” trademark/logo, the "Libre & Free" mention,
+// the words “You are using the Free and Open Source version of LinShare™, powered by
+// Linagora © 2009–2020. Contribute to Linshare R&D by subscribing to an Enterprise
+// offer!”. You must also retain the latter notice in all asynchronous messages such as
+// e-mails sent with the Program, (ii) retain all hypertext links between LinShare and
+// http://www.linshare.org, between linagora.com and Linagora, and (iii) refrain from
+// infringing Linagora intellectual property rights over its trademarks and commercial
+// brands. Other Additional Terms apply, see
+// <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf>
+// for more details.
+// This program is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+// more details.
+// You should have received a copy of the GNU Affero General Public License and its
+// applicable Additional Terms for LinShare along with this program. If not, see
+// <http://www.gnu.org/licenses/> for the GNU Affero General Public License version
+//  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
+//  the Additional Terms applicable to LinShare software.
+
+import 'package:dartz/dartz.dart';
+import 'package:domain/domain.dart';
+import 'package:domain/src/repository/document/document_repository.dart';
+
+class DownloadFileInteractor {
+  final DocumentRepository _documentRepository;
+  final TokenRepository _tokenRepository;
+  final CredentialRepository _credentialRepository;
+
+  DownloadFileInteractor(this._documentRepository, this._tokenRepository, this._credentialRepository);
+
+  Future<Either<Failure, Success>> execute(DocumentId documentId) async {
+    try {
+      final token = await _tokenRepository.getToken();
+      final baseUrl = await _credentialRepository.getBaseUrl();
```

Comment:
2 task can be run parallel. How we can do with await/async? @manhduc-nguyen

Reply:
What do you mean ? It's already async isn't it ?

Reply:
are you sure? IMO, `getBaseUrl` need to waiting for `getToken`

Reply:
Ok I see

Reply:
@hoangdat thanks for pointing this out, you are right. @imGok  I suggest you use this way to have `parallel execution`.
[https://stackoverflow.com/questions/55502528/flutter-multiple-async-methods-for-parrallel-execution](https://stackoverflow.com/questions/55502528/flutter-multiple-async-methods-for-parrallel-execution)

Reply:
I don't like `wait` because it retrieves a list of response and we can't get one item without casting each item

Reply:
Tell me what do you think of this version, If you agree I'll change it for remaining interactors

Reply:
no, please focus on this ticket only. other interactors, please do in the future.